utils/schemed-kafka-components-utils/src/test/scala/pl/touk/nussknacker/engine/schemedkafka/encode/BestEffortJsonSchemaEncoderTest.scala

```diff
@@ -229,6 +229,37 @@ class BestEffortJsonSchemaEncoderTest extends AnyFunSuite {
     new BestEffortJsonSchemaEncoder(ValidationMode.strict).encode(Map("foo" -> "bar", "redundant" -> 15), rejectAdditionalProperties) shouldBe 'invalid
   }
 
+  // delete above test if this one passes
+  ignore("should accept and encode redundant parameters if validation modes allows this") {
+    val allowAdditionalProperties: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$allowAdditionalProperties": "https://json-schema.org/draft-07/schema",
+        |  "type": "object",
+        |  "properties": {
+        |    "foo": {
+        |      "type": "string"
+        |    }
+        |  }
+        |}""".stripMargin))
+
+    val rejectAdditionalProperties: Schema = SchemaLoader.load(new JSONObject(
+      """{
+        |  "$allowAdditionalProperties": "https://json-schema.org/draft-07/schema",
+        |  "type": "object",
+        |  "properties": {
+        |    "foo": {
+        |      "type": "string"
+        |    }
+        |  },
+        |  "additionalProperties": false
```

Comment:
What about validation mode? Have you decided how it should work with `additionalProperties`?